### PR TITLE
Add list all Domain datasources

### PR DESCRIPTION
**Pull request by davidbloss**
## Issues

[Write first datasource with tests](https://github.com/OpsLevel/team-platform/issues/252)

## Changelog

- Add `DomainDataSources` for listing all domains in an account
Testing needs work...

- [X] List your changes here
- [ ] Make a `changie` entry

## Tophatting

Tested against `orange`...

```tf
# workspace/main.tf
data "opslevel_domain" "example" {
  identifier = "my_domain"
}

data "opslevel_domains" "all" {}

output "found" {
  value = data.opslevel_domain.example
}

output "found_all" {
  value = data.opslevel_domains.all
}
```

Run `task plan`
```tf
data.opslevel_domain.example: Reading...
data.opslevel_domains.all: Reading...
data.opslevel_domain.example: Read complete after 1s [id=Z2lkOi8vb3BzbGV2ZWwvRW50aXR5T2JqZWN0LzUyNDg0Nw]
data.opslevel_domains.all: Read complete after 1s

Changes to Outputs:
  + found     = {
      + aliases     = [
          + "my_domain",
        ]
      + description = "Hello World Domain"
      + id          = "Z2lkOi8vb3BzbGV2ZWwvRW50aXR5T2JqZWN0LzUyNDg0Nw"
      + identifier  = "my_domain"
      + name        = "My Domain"
      + owner       = "Z2lkOi8vb3BzbGV2ZWwvVGVhbS85NzU5"
    }
  + found_all = {
      + domains = [
          + {
              + aliases     = [
                  + "my_domain",
                ]
              + description = "Hello World Domain"
              + id          = "Z2lkOi8vb3BzbGV2ZWwvRW50aXR5T2JqZWN0LzUyNDg0Nw"
              + name        = "My Domain"
              + owner       = "Z2lkOi8vb3BzbGV2ZWwvVGVhbS85NzU5"
            },
          + {
              + aliases     = [
                  + "my_domain_2",
                ]
              + description = ""
              + id          = "Z2lkOi8vb3BzbGV2ZWwvRW50aXR5T2JqZWN0LzgxNDEzOQ"
              + name        = "My Domain"
              + owner       = ""
            },
          + {
              + aliases     = [
                  + "my_domain_3",
                ]
              + description = ""
              + id          = "Z2lkOi8vb3BzbGV2ZWwvRW50aXR5T2JqZWN0LzgxNDE0Mw"
              + name        = "My Domain"
              + owner       = ""
            },
          + {
              + aliases     = [
                  + "epic_2",
                ]
              + description = "epic2"
              + id          = "Z2lkOi8vb3BzbGV2ZWwvRW50aXR5T2JqZWN0LzEwOTE3ODE"
              + name        = "epic"
              + owner       = ""
            },
            ... 
You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.
```

### CODE REVIEW

File: opslevel/datasource_opslevel_domains.go [hunk 1]
```diff
@@ -1,9 +1,156 @@
...
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	domains, err := d.client.ListDomains(nil)
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read example, got error: %s", err))
```
Comment (rocktavious) [inline opslevel/datasource_opslevel_domains.go:107]: should we just pass the `err` into this?

File: opslevel/datasource_opslevel_domains.go [hunk 2]
```diff
@@ -1,9 +1,156 @@
...
+	data.Names = domainNames
+	resp.Diagnostics.Append(diag...)
+
+	domainDescriptions, diag := types.ListValueFrom(ctx, types.StringType, descriptions)
+	data.Descriptions = domainDescriptions
+	resp.Diagnostics.Append(diag...)
+
+	domainOwners, diag := types.ListValueFrom(ctx, types.StringType, owners)
+	data.Owners = domainOwners
+	resp.Diagnostics.Append(diag...)
```
Comment (rocktavious) [inline opslevel/datasource_opslevel_domains.go:145]: 🤯 - so strange

File: opslevel/provider.go [hunk 3]
```diff
@@ -134,14 +136,11 @@ func (p *OpslevelProvider) Configure(ctx context.Context, req provider.Configure
...
 func (p *OpslevelProvider) Resources(context.Context) []func() resource.Resource {
-	return []func() resource.Resource{
-		NewDomainResource,
-	}
+	return []func() resource.Resource{}
 }
 
 func (p *OpslevelProvider) DataSources(context.Context) []func() datasource.DataSource {
 	return []func() datasource.DataSource{
-		NewDomainDataSource,
```
Comment (rocktavious) [inline opslevel/provider.go:144]: seems like the base branch needs to be fixed

File: opslevel/datasource_opslevel_domains.go (hunk 1 above)
Comment (davidbloss) [reply opslevel/datasource_opslevel_domains.go:107]: Not here. This func expects two string args:
`func (diags *diag.Diagnostics) AddError(summary string, detail string)`

**Review by taimoorgit — APPROVED** [review]

**Review by taimoorgit — APPROVED** [review]

**Pull request merged**